pkg/gardenlet/controller/seed/seed.go

```diff
@@ -245,33 +246,46 @@ func (c *Controller) startHealthManagement(ctx context.Context) {
 		default:
 			time.Sleep(LeaseResyncGracePeriodSeconds / 2 * time.Second)
```

Comment:
As you're already on it, can you change this to a non-blocking case?
```suggestion
		case <-time.After(LeaseResyncGracePeriodSeconds / 2 * time.Second):
```